### PR TITLE
SwapNFT contract

### PR DESCRIPTION
**Pull request by 0xcdb**

### CODE REVIEW

File: contracts/SwapNFT.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    function transferFrom(address, address, uint256) public pure override {
+        revert NotTransferrable();
+    }
+
+    function safeTransferFrom(address, address, uint256) public pure override {
+        revert NotTransferrable();
+    }
+
+    function safeTransferFrom(address, address, uint256, bytes memory) public pure override {
```
Comment (alextheboredape) [inline contracts/SwapNFT.sol:39]: safeTransferFrom is twice. we need to fix this and add transfer and safeTransfer

File: contracts/SwapNFT.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+contract SwapNFT is ERC721, ISwapNFT {
+    address public immutable aelin;
+
+    uint256 public tokenCount;
+
+    mapping(address => bool) public hasMinted;
+
+    constructor(address _aelinAddress) ERC721("AelinSwapNFT", "SwapNFT") {
+        aelin = _aelinAddress;
```
Comment (alextheboredape) [inline contracts/SwapNFT.sol:17]: we can hardcode this value instead of passing it in

File: contracts/SwapNFT.sol [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    function mint() external {
+        if (IERC20(aelin).balanceOf(msg.sender) == 0) revert BalanceIsZero();
```
Comment (alextheboredape) [inline contracts/SwapNFT.sol:21]: we can mint them a NFT even if their balance is 0. they can get the AELIN later. I am ok with keepign this check though if you want

File: contracts/SwapNFT.sol (hunk 1 above)
Comment (0xcdb) [reply contracts/SwapNFT.sol:39]: - they are not the same

- transfer and safeTransfer aren't public by default according to ERC721 transfer so nothing to do here

File: contracts/SwapNFT.sol (hunk 2 above)
Comment (0xcdb) [reply contracts/SwapNFT.sol:17]: it's fine to leave it like this in case we want to deploy on testnet

File: contracts/SwapNFT.sol (hunk 3 above)
Comment (0xcdb) [reply contracts/SwapNFT.sol:21]: it's actually a good point. I will remove this line

File: contracts/AelinBurner.sol [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    bool public nftSet;
+
+    // TODO create the contract from the deployer and call transferOwnership to the multisig
+    // before sending the assets to the contract
+    constructor() {
+        start = block.timestamp;
+    }
+
+    function setNft(address _swapNFT) external onlyOwner {
+        if (nftSet) revert NftAlreadySet();
```
Comment (0xcdb) [inline contracts/AelinBurner.sol:43]: we don't need `nftSet`. We can just say `if (swapNFT != address(0)) revert`

File: contracts/AelinBurner.sol [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    address public constant USDC = 0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85;
+    address public constant VEKWENTA = 0x678d8f4Ba8DFE6bad51796351824DcCECeAefF2B;
+
+    uint256 public constant WITHDRAW_WINDOW = 1 weeks;
+
+    address public swapNFT;
+    bool public nftSet;
+
+    // TODO create the contract from the deployer and call transferOwnership to the multisig
+    // before sending the assets to the contract
```
Comment (0xcdb) [inline contracts/AelinBurner.sol:37]: this probably doesn't belong here

**Pull request merged**